sdk/src/asset_handlers/jpeg_io.rs

```diff
@@ -408,22 +438,27 @@ impl AssetIO for JpegIO {
     }
 
     fn save_cai_store(&self, asset_path: &std::path::Path, store_bytes: &[u8]) -> Result<()> {
-        let mut stream = std::fs::OpenOptions::new()
+        let mut input_stream = std::fs::OpenOptions::new()
             .read(true)
-            .write(true)
             //.truncate(true)
             .open(asset_path)
             .map_err(Error::IoError)?;
 
-        self.write_cai(&mut stream, store_bytes)?;
+        let mut temp_file = Builder::new()
```

Comment:
We probably need to centralize our temp file usage in some utility. I know there can be issues, and having it implemented in scattered places will make it harder to debug issues.